### PR TITLE
Ansible

### LINKED ISSUE #562: Provide ansible role for Kubeone
**What feature would you like to be added?**
Add an ansible role that provides access to the features KubeOne is offering.

**What are use cases of the feature?**
People using ansible have a nicer access to KubeOne

### PR DESCRIPTION
**Pull request by eqrx**
**What this PR does / why we need it**:
Brings roles for Ansible installation and upgrade

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #562

**Special notes for your reviewer**:
None

**Does this PR introduce a user-facing change?**:
```release-note
We now have Ansible roles and examples 
```

### CODE REVIEW

**Comment by eqrx** [comment]
Blocked by https://github.com/kubermatic/kubeone/pull/575

File: examples/ansible/roles/kubeone-upgrade/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+kubeone_clusterNetwork_cni_encrypted: false
+kubeone_cloudProvider_name: none
+kubeone_cloudProvider_external: false
+kubeone_cloudProvider_cloudconfig: ""
+kubeone_apiEndpoint_host: ""
+kubeone_apiEndpoint_port: 0
+kubeone_features_podSecurityPolicy_enable: false
+kubeone_features_dynamicAuditLog_enable: false
+kubeone_features_metricsServer_enable: true
+kubeone_machineController_deploy: true
```
Comment (xmudrii) [inline examples/ansible/roles/kubeone-upgrade/defaults/main.yml]: If you add the missing `machineController` section and set this to `false` (assuming you don't want machine-controller), KubeOne is going to work even without the fix. :slightly_smiling_face:

File: README.md [hunk 2]
```diff
@@ -172,7 +172,7 @@ about the infrastructure.
 kubeone install config.yaml --tfjson tf.json
 ```
 
-To learn more about KubeOne configuration, please run `kubeone config print --full`.
+To learn more about KubeOne configuration, please run `kubeone config print --full`. You may also use our [Ansible roles][12].
```
[the remark this answers is not in the document]
Comment (eqrx) [reply README.md]: Mh, could you elaborate? I shuffled the sentence around so it should be more clear

File: examples/ansible/roles/kubeone-upgrade/defaults/main.yml (hunk 1 above)
Comment (eqrx) [reply examples/ansible/roles/kubeone-upgrade/defaults/main.yml]: jup

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (xmudrii) [reply README.md]: Looks great :100:

**Pull request merged**